Tests/BUILD

```diff
@@ -9,11 +9,18 @@ config_setting(
     constraint_values = ["@platforms//os:linux"],
 )
 
+genrule(
+    name = "LinuxMain",
+    srcs = ["LinuxMain.swift"],
+    outs = ["main.swift"],
+    cmd = "cp $< $@"
+)
```

Comment:
@Keith without this change Linux builds do not work, how did you get this working on other projects?

Reply:
I'd have to debug a bit but we use this macro which definitely outputs it as LinuxMain.swift:


```bzl
load("@build_bazel_rules_swift//swift:swift.bzl", "swift_library", "swift_test")

def swiftpm_test(
        name,
        srcs,
        deps,
        data = [],
        test_tags = []):
    """Emulate Swift Package Manager's test target behavior

    Args:
        name: The name of the test target
        srcs: A list of source files to compile for the tests
        deps: A list of targets the test module depends on
        data: Any resources that should be made available at test time
        test_tags: Additional tags to pass to the test target
    """
    library_name = name + ".library"
    swift_library(
        name = library_name,
        srcs = srcs,
        data = data,
        deps = deps,
        module_name = name,
    )

    rule_name = "{}.linuxmain".format(name)
    test_srcs = [rule_name]
    native.genrule(
        name = rule_name,
        srcs = srcs,
        cmd = "./$(location //tools:linux_main_generator) {} $(SRCS) > $(OUTS)".format(name),
        outs = ["Tests/LinuxMain.swift"],
        exec_tools = ["//tools:linux_main_generator"],
        message = "Generating LinuxMain.swift",
    )

    swift_test(
        name = name,
        srcs = select({"//:linux": test_srcs, "//conditions:default": []}),
        deps = [library_name],
        tags = test_tags,
        size = "small",
    )
```